saleor/order/calculations.py

```diff
@@ -428,10 +431,19 @@ def _apply_tax_data(
         )
         undiscounted_subtotal += order_line.undiscounted_total_price
 
+        if not order_line.is_gift:
+            order_line.unit_discount_amount = quantize_price(
+                order_line.undiscounted_unit_price_net_amount
+                - order_line.unit_price_net_amount,
+                currency,
+            )
+
```

Comment:
Also, I might be wrong, but I don't see that we call `_apply_tax_data(` for plugins. Which means, that these data will not be propagated there. (This is at least from what I see in the code).

Reply:
For plugins we use the same logic as for flat rates, so the `unitDiscountReason` and `unitDiscountValue` is updated in `base_calculation.apply_order_discounts`